src/main/kotlin/org/danilopianini/plagiarismdetector/analyzer/representation/token/LanguageTokenTypesImpl.kt

```diff
@@ -0,0 +1,15 @@
+package org.danilopianini.plagiarismdetector.analyzer.representation.token
+
+/**
+ * A simple class implementing a [LanguageTokenTypes].
+ * @param tokensTypes the set of token types of the language
+ */
+class LanguageTokenTypesImpl(private val tokensTypes: Collection<TokenType>) : LanguageTokenTypes {
+    override fun tokenFor(constructName: String): TokenType {
+        require(isToken(constructName))
+        return tokensTypes.find { it.languageConstructs.contains(constructName) }!!
```

Comment:
Why the non-null assertion? This is NPE-prone.

Reply:
```kotlin
override fun tokenFor(constructName: String): TokenType =
    tokensTypes.find { constructName in it.languageConstructs } ?: error("No token found for given construct name.")
```

Better?

Reply:
A lot. And even better with a cleaner error:
`error("Token $constructName does not exist. Available constructs: $tokensTypes")`

Failing clearly is very helpful.